client/src/app/containers/Home/ActiveGamesTable.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { useInjectReducer, useInjectSaga } from 'redux-injectors';
+import { selectCurrentUser } from '../../../auth/selectors';
+import { authSaga } from '../../../auth/saga';
+import {
+  sliceKey as authSliceKey,
+  reducer as authReducer,
+} from '../../../auth/slice';
+import { selectActiveGames } from './selectors';
+import { homeSaga } from './saga';
+import {
+  sliceKey as homeSliceKey,
+  reducer as homeReducer,
+  actions as homeActions,
+} from './slice';
+
+const ActiveGamesTable = () => {
+  useInjectReducer({ key: authSliceKey, reducer: authReducer });
+  useInjectSaga({ key: authSliceKey, saga: authSaga });
+  useInjectReducer({ key: homeSliceKey, reducer: homeReducer });
+  useInjectSaga({ key: homeSliceKey, saga: homeSaga });
+  const dispatch = useDispatch();
+  const currentUser = useSelector(selectCurrentUser);
+  const activeGames = useSelector(selectActiveGames(currentUser.id));
+
+  // event handlers
+  const onRefreshActiveGames = () => {
+    dispatch(homeActions.refreshActiveGames({ id: currentUser.id }));
+  };
+
+  let gameButtons = activeGames.map(activeGame => {
+    if (!activeGame || !activeGame.gameID) {
+      return;
+    }
+    const gID = activeGame.gameID;
+
+    let opponents = [];
```

Comment:
Sorry. I might've commented this on #71. I confuse myself. But can we have the server do the heavy lifting here and return us `opponents` in the response?

Reply:
The server returns a map of all playerIDs to player's name in the game. How the client interprets that will be up to the location. In this place, we only want the opponent's names, so we filter out "my name" from the list. I suppose I should use a filter func instead because that's the JS way to do things

Reply:
This also won’t work in 4 player games because there will be one other player who isn’t an opponent.

Does it hurt us to have the server return a list of opponents? Or can we have an endpoint for getting opponents? Just tryna make it easy for the client to do it’s primary job well: display and allow the user to interact with the data.